src/rust/engine/logging/src/logger.rs

```diff
@@ -85,27 +88,30 @@ impl Logger {
     python_level: u64,
   ) -> Result<std::os::unix::io::RawFd, String> {
     use std::os::unix::io::AsRawFd;
-    python_level.try_into().and_then(|level: PythonLogLevel| {
-      {
-        // Maybe close open file by dropping the existing logger
-        *self.pantsd_log.lock() = MaybeWriteLogger::empty();
-      }
-      OpenOptions::new()
-        .create(true)
-        .append(true)
-        .open(log_file_path)
-        .map(|file| {
-          let fd = file.as_raw_fd();
-          self.maybe_increase_global_verbosity(level.into());
-          *self.pantsd_log.lock() = MaybeWriteLogger::new(
-            file,
-            level.into(),
-            self.show_rust_3rdparty_logs.load(Ordering::SeqCst),
-          );
-          fd
-        })
-        .map_err(|err| format!("Error opening pantsd logfile: {}", err))
-    })
+    python_level
+      .try_into()
+      .map_err(|err| format!("{}", err))
```

Comment:
We do this so often in the pants codebase that I'd love to have a macro for it. Idle thoughts.